### PR TITLE
[ONNX] Moved uninitialized_optional test cases into runtime test and parameterized left test cases

### PR DESCRIPTION
**Pull request by titaiwangms**
- moved test_uninitialized_optional into runtime test, because **OptionalHasElement** node now is supported on onnxruntime. test_optional_input will be moved later when tracing/None output paradox is fixed.
- changed _TestONNXRuntime to inherited by common_utils.Testcase as it's more rich on error message
- modified the left itertools implemented test cases to using parametrized decorator

### CODE REVIEW

File: test/onnx/test_pytorch_onnx_onnxruntime.py [hunk 1]
```diff
@@ -12214,7 +12214,20 @@ def forward(self, x):
         self.run_test(Module(True), x, rtol=1e-3, atol=1e-6)
 
     @skipIfUnsupportedMinOpsetVersion(16)
-    def test_grid_sample(self):
+    @common_utils.parametrize(
+        "mode",
+        ("bilinear", "nearest", "bicubic"),
+        name_fn=lambda mode: mode,
```
Comment (justinchuby) [inline test/onnx/test_pytorch_onnx_onnxruntime.py:12220]: According to the doc, we don't need the name_fn for simple cases: https://sourcegraph.com/github.com/pytorch/pytorch@bcc02769bef1d7b89bec724223284958b7c5b564/-/blob/torch/testing/_internal/common_utils.py?L248:7&subtree=true

**Review by justinchuby — APPROVED** [review]
I would: (1) update the PR title to be more specific on the change and (2) include _why_ the change was needed in the PR description.

File: test/onnx/test_pytorch_onnx_onnxruntime.py (hunk 1 above)
Comment (AllenTiTaiWang) [reply test/onnx/test_pytorch_onnx_onnxruntime.py]: You are right. Only leaving **align_corners** as it would be ambiguous without naming.

**Review by BowenBao — APPROVED** [review]
LGTM, feel free to merge after resolving lintrunner error.

**Pull request closed without merging**